### PR TITLE
autofill entity of new repeated set

### LINKED ISSUE #12063: [Enumerators] Pre-populate admin id and question text for a new enumerator question
#### Describe the solution presented in this issue.
Pre-populate admin id and question text fields on the new enumerator creation form on the Admin Program Block edit page once the repeated entity name is filled out.  

The question text will follow this format:  "Please add all your [listed entity]s:"  We'll replace "[listed entity] with the repeated entity they enter in the previous field.

#### Additional Context
See [this section of the TDD ](https://docs.google.com/document/d/1LKcPuPc6ruRyk5F5oBbi6JaQb_EpS4RBlZULilFEWfI/edit?tab=t.0#heading=h.a8qkblkfmtbp)for implementation details.

### PR DESCRIPTION
**Pull request by CallaJune**
### Description

Pre-populate admin ID and question text in new enum question.

Screenshot 2026-03-24 at 1 42 52 PM

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Added the correct label (see [docs](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-appropriate-labels) for more info): < feature | enhancement | bug | under-development | dependencies | infrastructure | ignore-for-release | database >
- [x] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers)
- [ ] Added an additional reviewer from `civiform/eng-admin` as FYI (if this PR includes functionality changes and neither you nor the primary reviewer is an admin)
- [x] Removed the release notes section if the title is sufficient for the release notes description, or put more details in that section.
- [x] Created unit and/or browser tests which fail without the change (if possible)
- [x] Performed manual testing (Chrome and Firefox if it includes front-end changes)
- [x] Extended the README / documentation, if necessary. For user-facing features, consider updating [the user docs](https://github.com/civiform/docs). For "under-the-hood" changes or things more relevant to developers, consider updating [the dev wiki](https://github.com/civiform/civiform/wiki).
- [x] Ensured PII wasn't added to any new logs, unless it was guarded by `isDevOrStaging`
- [x] Noted in the PR description which, if any, code in this PR was generated by AI.

### Instructions for manual testing

Scenario A: Basic Autofill
1. Create a new program or open an existing one
2. Add a new Repeated Set Block
3. Fill the "Repeated set name" field with something like household member
4. Admin ID field should auto-populate with: household member repeated set. Question text field should auto-populate with: Please add each household member

Scenario B: Preserve Manual Edits
1. In the same form, manually change:
2. Admin ID to: my custom id
3. Question text to: My custom question
4. Now change the "Repeated set name" to income source
5. Verify: Your manual edits are preserved (fields don't change). 
6. Clear the Admin ID field (make it empty)
7. Change "Repeated set name" to household item
8. Verify: Admin ID auto-fills again with household item repeated set because the field was empty

### Issue(s) this completes

Fixes #12063

### CODE REVIEW

**Comment by CallaJune** [comment]
* **#13016** <a href="https://app.graphite.com/github/pr/civiform/civiform/13016?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/civiform/civiform/13016?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: browser-test/src/end_to_end_enumerators.test.ts [hunk 1]
```diff
@@ -808,6 +808,68 @@ test.describe('End to end enumerator test with enumerators feature flag on', ()
...
+        await expect(
+          blockPanel.getByRole('textbox', {name: 'Repeated set admin ID'}),
+        ).toHaveValue('household member repeated set')
+        await expect(
+          blockPanel.getByRole('textbox', {name: 'Question text'}),
+        ).toHaveValue('Please add each household member')
+      })
+
+      await test.step('Preserve manual edits for suggested fields', async () => {
+        const adminIdInput = blockPanel.getByRole('textbox', {
```
Comment (melanie-exygy) [inline browser-test/src/end_to_end_enumerators.test.ts]: **suggestion (non-blocking):** If you create this constant before the 2nd step of the test (outside the step), you'll be able to re-use it, for example in the expectation on line 829.

File: browser-test/src/end_to_end_enumerators.test.ts [hunk 2]
```diff
@@ -808,6 +808,68 @@ test.describe('End to end enumerator test with enumerators feature flag on', ()
...
+          name: 'Repeated set admin ID',
+        })
+        await adminIdInput.fill('custom repeated set id')
+
+        const questionTextInput = blockPanel.getByRole('textbox', {
```
Comment (melanie-exygy) [inline browser-test/src/end_to_end_enumerators.test.ts]: **suggestion (non-blocking):** Same suggestion as above.

File: browser-test/src/end_to_end_enumerators.test.ts [hunk 3]
```diff
@@ -808,6 +808,68 @@ test.describe('End to end enumerator test with enumerators feature flag on', ()
...
+
+        await expect(
+          blockPanel.getByRole('textbox', {name: 'Repeated set admin ID'}),
+        ).toHaveValue('household member repeated set')
+        await expect(
+          blockPanel.getByRole('textbox', {name: 'Question text'}),
+        ).toHaveValue('Please add each household member')
+      })
+
+      await test.step('Preserve manual edits for suggested fields', async () => {
```
Comment (melanie-exygy) [inline browser-test/src/end_to_end_enumerators.test.ts:841]: **praise:** Thanks for thinking of this user scenario!  It's something I hadn't considered.

File: browser-test/src/end_to_end_enumerators.test.ts [hunk 4]
```diff
@@ -869,16 +935,9 @@ test.describe('End to end enumerator test with enumerators feature flag on', ()
       })
 
       await test.step('Submit the new enumerator question form with a duplicate admin ID', async () => {
-        await blockPanel
```
Comment (melanie-exygy) [inline browser-test/src/end_to_end_enumerators.test.ts:872]: **praise:** Thanks for cleaning this up!

File: server/app/assets/javascripts/admin_programs.ts [hunk 5]
```diff
@@ -16,6 +16,14 @@ class AdminPrograms {
...
   private static DISABLED_TEXT_CLASS = 'read-only:text-gray-500'
   private static DISABLED_BACKGROUND_CLASS = 'read-only:bg-gray-100'
+  private static REPEATED_SET_LISTED_ENTITY_INPUT_ID = 'listed-entity-input'
+  private static REPEATED_SET_ADMIN_ID_INPUT_ID = 'enumerator-admin-id-input'
+  private static REPEATED_SET_QUESTION_TEXT_INPUT_ID = 'question-text-input'
+  private static REPEATED_SET_PREVIOUS_ADMIN_ID_SUGGESTION_DATA_ATTRIBUTE =
+    'previousAdminIdSuggestion'
+  private static REPEATED_SET_PREVIOUS_QUESTION_TEXT_SUGGESTION_DATA_ATTRIBUTE =
+    'previousQuestionTextSuggestion'
+  private static hasAttachedRepeatedSetAutofillListener = false
```
Comment (melanie-exygy) [inline server/app/assets/javascripts/admin_programs.ts]: **nitpick:** seems like this should be screaming snake case to be consistent with the other static fields

File: server/app/assets/javascripts/admin_programs.ts [hunk 6]
```diff
@@ -403,6 +412,96 @@ class AdminPrograms {
...
+    if (this.hasAttachedRepeatedSetAutofillListener) {
+      return
+    }
+
+    document.body.addEventListener('input', (event) => {
+      const target = event.target as HTMLElement | null
+      if (!(target instanceof HTMLInputElement)) {
+        return
+      }
+      if (target.id !== this.REPEATED_SET_LISTED_ENTITY_INPUT_ID) {
```
Comment (melanie-exygy) [inline server/app/assets/javascripts/admin_programs.ts]: **suggestion (non-blocking):** Rather than adding an event listener for every input event on the document body, I think you could just add a listener to the listed entity input element, like this:

```
const listedEntityInputElement = document.getElementById(this.REPEATED_SET_LISTED_ENTITY_INPUT_ID);

listedEntityInputElement.addEventListener('input', ...
```

File: server/app/assets/javascripts/admin_programs.ts [hunk 7]
```diff
@@ -403,6 +412,96 @@ class AdminPrograms {
...
+        return
+      }
+
+      this.maybeAutofillRepeatedSetFormFields(target)
+    })
+
+    this.hasAttachedRepeatedSetAutofillListener = true
+    this.maybeAutofillRepeatedSetFormFields()
```
Comment (melanie-exygy) [inline server/app/assets/javascripts/admin_programs.ts]: **question (non-blocking):** Why do we need to autofill on init?  Is it in case the listed entity field loads already populated?  If so, that makes sense, I just want to make sure I understand.

File: server/app/assets/javascripts/admin_programs.ts [hunk 8]
```diff
@@ -351,6 +359,7 @@ class AdminPrograms {
         return
       }
       if (targetElement.id === 'enumerator-setup') {
+        this.maybeAutofillRepeatedSetFormFields()
```
Comment (melanie-exygy) [inline server/app/assets/javascripts/admin_programs.ts]: **question:** I'm not sure that we should autofill on error because if the user has left the admin id blank and they get the error, the field will be filled for them and they may be confused as to why they got the "admin id can't be blank" error.  Would you mind checking with Jia?  I can demo the situation I'm thinking of if it isn't clear.

File: server/conf/i18n/messages [hunk 9]
```diff
@@ -1169,6 +1169,10 @@ input.repeatedSet.listedEntity=Listed entity
 input.repeatedSet.adminId=Repeated set admin ID
 # A text input field asking admins for the text that will prompt applicants to list out the entity
 input.repeatedSet.questionText=Question text
+# Autofill suggestion template shown in the admin ID field when applicant selects a listed entity. {0} is the listed entity name.
+input.repeatedSet.adminIdAutofillSuggestion={0} repeated set
+# Autofill suggestion template shown in the question text field when applicant selects a listed entity. {0} is the listed entity name.
+input.repeatedSet.questionTextAutofillSuggestion=Please add each {0}
```
Comment (melanie-exygy) [inline server/conf/i18n/messages]: **suggestion (blocking):** Could you add a period at the end?

File: server/app/assets/javascripts/admin_programs.ts (hunk 8 above)
Comment (melanie-exygy) [reply server/app/assets/javascripts/admin_programs.ts]: After talking with Jia, we decided not to autofill on error because the user would have consciously removed or changed the admin id or question text to get to the error situation.  We don't want to override their decision.

File: browser-test/src/support/admin_program_image.ts
Comment (melanie-exygy) [inline browser-test/src/support/admin_program_image.ts:1]: **note:** This is just the formatter doing its thing.

File: browser-test/src/end_to_end_enumerators.test.ts [hunk 10]
```diff
@@ -808,6 +808,59 @@ test.describe('End to end enumerator test with enumerators feature flag on', ()
       })
     })
 
+    test('auto-fills and preserves editable repeated set suggestions', async ({
```
Comment (AlvieH) [inline browser-test/src/end_to_end_enumerators.test.ts:811]: Are we testing possible corner cases for the normalization for the repeated block entity? Otherwise looks good!

Comment (melanie-exygy) [reply browser-test/src/end_to_end_enumerators.test.ts:811]: Added some extra spaces in the listed entity to test for normalization.
